src/writers.ts

```diff
@@ -8,15 +16,29 @@ export function createEvmWriters(indexerName: string) {
 
     const sender = event.args.sender;
     const token = event.args.token;
-    const amount = event.args.amount;
+    const amount = (Number(event.args.amount.toString()) / 1e6).toFixed(2); // Transforms 1990000 to 1.99
     const barcode = event.args.barcode;
 
-    const payment = new Payment(`${sender}/${tx.hash}`, indexerName);
+    const chain = indexerName; // The indexer name corresponds to the chain name
+    let tokenName = getTokenName(token, chain);
+
+    if (!tokenName) {
+      // TODO log missing token
+      tokenName = token;
+    }
+
+    const payment = new Payment(`${tx.hash}`, indexerName);
     payment.sender = sender;
-    payment.token = token;
+    payment.token = tokenName;
     payment.amount = amount;
+
     payment.barcode = barcode;
+    const ipfsData = await getJSON(barcode);
+
     payment.block = block.number;
+    payment.product = ipfsData.type;
```

Comment:
Can we call it `type`? A payment may not always be related to a product, `type` is more general. 
```suggestion
    payment.type = ipfsData.type;
```

Reply:
321c57462185f4346328bf647c7a781c5a10bb7e